src/odemis/gui/util/filename_pattern.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 29 Jan 2018
+
+@author: Philip Winkler
+
+Copyright © 2018 Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+
+from __future__ import division
+import re
+import time
+import os
+from odemis.dataio import get_available_formats
+
+
+def generate_fn_ptn(fn):
```

Comment:
In general, avoid abbreviations in the function names. I'd suggest to rename to `guess_pattern()`, to highlight it's entirely sure it will find out the actual pattern expected by the user.